### PR TITLE
Simplify FormController by not tracking current action

### PR DESCRIPTION
**Pull request by imobachgs**
Always add nested elements and trust in the backup/restore mechanism.

### CODE REVIEW

**Comment by mvidner** [comment]
Let me describe how this works to see if I get it right:

The ControllerState stack describes what each open dialog is doing. Now, we're removing the *action* part, so the dialog no longer cares - adding an item is just like adding a default one, then editing it. Canceling works by the rollback mechanism.

File: src/lib/y2configuration_management/salt/form_controller.rb [hunk 1]
```diff
@@ -199,29 +207,17 @@ def run_popup(action, relative_locator)
...
-        end
+        form_data.update(state.locator, result)
       end
 
       # Adds or updates the parent of a collection item
       #
       # When trying to add an element to a collection, it is necessary that the parent
       # object exists.
-      def add_or_update_parent
+      def update_parent
```
Comment (mvidner) [inline src/lib/y2configuration_management/salt/form_controller.rb]: Please adjust the method comment

File: src/lib/y2configuration_management/salt/form_controller.rb [hunk 2]
```diff
@@ -199,29 +207,17 @@ def run_popup(action, relative_locator)
       # @param result [Hash,nil] Result to process
       def update_form_data(result)
```
Comment (mvidner) [inline src/lib/y2configuration_management/salt/form_controller.rb]: As the return value is unused anyway, I'd suggest inlining this method if we're going to bring `open_form` and `close_form` together

File: src/lib/y2configuration_management/salt/form_controller.rb [hunk 3]
```diff
@@ -232,9 +200,31 @@ def item_form_for(locator)
...
       end
 
+      # Finds or creates the item to be edited
+      #
+      # @return [Hash]
+      def find_or_create_item(item_locator)
+        new_item = get(item_locator)
+        return new_item if new_item
+        form_data.add_item(item_locator.parent, {})
+        {}
```
Comment (mvidner) [inline src/lib/y2configuration_management/salt/form_controller.rb]: wait, should we really return a different empty hash than the one we've added?

Comment (imobachgs) [reply src/lib/y2configuration_management/salt/form_controller.rb]: You are right.

**Pull request merged**